### PR TITLE
Basic OpenVR Support code

### PR DESCRIPTION
**Pull request by jamesu**
Adds basic OpenVR support. This also includes support for rendering controllers based on OpenVR data, though this is as of yet incomplete as there are no hooks for the button input.

Much like the oculus support, a script is also provided to handle setting up control input. However this is not hooked in by default for this PR.

The intention of this PR is more to add the support code to the engine and as a basis for identifying regressions with future dx11/gl changes.

### CODE REVIEW

File: Engine/source/T3D/player.h [hunk 1]
```diff
@@ -39,6 +39,7 @@ class DecalData;
 class SplashData;
 class PhysicsPlayer;
 class Player;
+class OpenVRTrackedObject;
```
Comment (Azaezel) [inline Engine/source/T3D/player.h]: Causes non-compilation with a default-selection build. Initial suggested resolution:
#ifdef TORQUE_OPENVR
class OpenVRTrackedObject;
#else
class OpenVRTrackedObject : public GameBase
{
   typedef GameBase Parent;
};
#endif

**Comment by jamesu** [comment]
@Azaezel that sounds pretty bad since your giving the fake impression that OpenVRTrackedObject is actually implemented according to the dummy stuff in the header, which could cause problems depending on the order of includes.

Could you at least tell me what compiler your using and what error your getting?

**Comment by Azaezel** [comment]
vs2013, was spamming
 use of undefined type 'OpenVRTrackedObject' without a remout and
C2065: 'OpenVRTrackedObject' : undeclared identifier with as the first two quick 'fixes'

**Comment by Areloch** [comment]
Sorry for the huge wait time on testing this(had to move all the vive stuff around). I grabbed this, merged with devhead and it compiled on the first try, no problems.

I'll test it compiling in vs2013 next to make sure it's not an oddball compiler version issue.

**Comment by jamesu** [comment]
Rebased to latest dev branch

**Pull request merged**